### ISSUE TITLE
idea: assign CODEOWNERS by file

### ISSUE THREAD

**Issue by import-brain** [withfig/autocomplete#768]
### Sanity checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my feature

### Description:
In the autocomplete repo, would it be better to assign spec creators as codeowners for the spec they created?
E.g. you created the gh spec, you are added in CODEOWNERS for that specific file

I believe this would help other people, in addition to the creator, change and revise specs, because the creator will be pinged whenever there is a PR affecting the file, and they could offer their input.

**Comment by import-brain** [comment]
> Having to maintain the codeowners file honestly doesn't sound as the most appealing thing to me.

Maybe, we could have people add themselves to CODEOWNERS in the same PR when they add new specs?

**Comment by fedeci** [comment]
Let's wait to hear what @mschrage and @QuiiBz think.

**Comment by mschrage** [comment]
I really like this idea! I think it empowers contributors and will keep people more involved.

I wonder if this can be done with a GitHub action?

**Comment by import-brain** [comment]
> I really like this idea! I think it empowers contributors and will keep people more involved.
> 
> I wonder if this can be done with a GitHub action?

It probably could be done with an action, is there any variable you can call in the GitHub API for PRs that's something similar to "files_affected"?

**Comment by QuiiBz** [comment]
Not sure how this could be implemented with a GH action. Also, sometimes people who create a spec don't want to maintain it, so this would have to be optional.

**Comment by import-brain** [comment]
> Not sure how this could be implemented with a GH action. Also, sometimes people who create a spec don't want to maintain it, so this would have to be optional.

@mschrage, any ideas on how this can be implemented while making it optional? A GH action probably wouldn't work.

**Comment by mschrage** [comment]
@import-brain @QuiiBz Could it be done as a precommit hook? 

If a new spec is staged that has no codeowner, the commit hook can ask the user to enter their Github username if they would like to become the owner of the spec.

**Comment by fedeci** [comment]
I think we can do something with fig-bot instead in the PR body.

**Comment by import-brain** [comment]
> I think we can do something with fig-bot in the PR body instead.

Yeah, perhaps a checkbox? That could work

**Comment by mschrage** [comment]
@import-brain would you be interested in implementing this?

**Comment by import-brain** [comment]
> @import-brain would you be interested in implementing this?

This is neither my area of expertise nor my cup of tea, is there anyone that knows how to implement it well? Who wrote the bot?

**Comment by fedeci** [comment]
I think Tim wrote it, right @mschrage?

**Comment by mschrage** [comment]
Yeah! But I'm not sure if he has time to work on this unfortunately.

**Comment by import-brain** [comment]
> Yeah! But I'm not sure if he has time to work on this unfortunately.

Well then, perhaps I could give it a try. With some experimentation, I could probably whip something up that works, might take a decent bit of time though...

**Comment by fedeci** [comment]
I can do it eventually.

**Comment by import-brain** [comment]
> I can do it eventually, but it wouldn't be my priority.

Sounds good to me

**Comment by BogDAAAMN** [comment]
Ohhhhh I like this, I would really love to do this. I can prototype a GitHub action tonight and show you, unless you folks @import-brain @fedeci wanna take your time with it

**Comment by fedeci** [comment]
Love it!🔥

**Comment by import-brain** [comment]
> Ohhhhh I like this, I would really love to do this. I can prototype a GitHub action tonight and show you, unless you folks @import-brain @fedeci wanna take your time with it

Great, please do it! I am not very experienced in actions, you'll do better than what I could do, at least :)

**Comment by fedeci** [comment]
@BogDAAAMN have you already started working on this? If not I can do it tomorrow.

**Comment by BogDAAAMN** [comment]
Noo, I totally overestimated my free time, I am sorry 😭 Started trying some stuff, but it's not worth checking out. Please do, I'd be happy to see what you do

**Comment by fedeci** [comment]
Drafted this https://github.com/fedeci/add-codeowners-action. It is a minimal version but it works, I will update it in the next days

**Comment by import-brain** [comment]
> Drafted this [fedeci/add-codeowners-action](https://github.com/fedeci/add-codeowners-action). It is a minimal version but it works, I will update it in the next days to support more features and to improve reliability

Wow, this looks great, thank you!

**Comment by import-brain** [comment]
Based on the fact that #806 was merged, I think this can be closed.

All, thank you for the great work! 🚀

**Comment by fedeci** [comment]
It was merged and reverted, I prefer to keep this open😅

**Comment by import-brain** [comment]
> It was merged and reverted, I prefer to keep this open😅

Didn't notice the revert PR, my bad 😅
